src/compilation/compiler.rs

```diff
@@ -337,26 +364,6 @@ impl Compiler {
         cur_func
     }
 
-    /// Compiles an assignment expression like `foo = 28` (and panics if `foo`'s type can't store 28).
-    unsafe fn compile_assignment(&mut self, assign: &Assign, expected_type: Type) -> LLVMValueRef {
-        let value = self.compile_expr(assign.value.as_ref(), expected_type);
-        let var = match self.scope_manager.get_var(&assign.name) {
-            Some(var) => var,
-            None => panic!("Setting a variable that has not been declared"),
-        };
-
-        if var.var_type != expected_type {
-            panic!(
-                "Assigning to variable '{}' with type '{}' but expected to have type '{}'",
-                assign.name, var.var_type, expected_type
-            );
-        }
```

Comment:
This PR removes type-checking during assignment, allowing stuff like:

```
i1 x;
x = 2;
``` 

I think we should let it slide for now, though, since we hadn't truly implemented type checking anywhere anyway. And we're about to.

Reply:
Ahh yes you're right. Good catch. The issue is we changed the assignment operation to be a statement instead of an expression, which means the "expected type" of "a = 2" is going to be null (since it shouldn't return anything). To add this check back (which we most definitely should) we need to be able to get the type of the right hand expression, which we will be able to do after type checking.